andela/ah-backend-thanos
# #161348794 Password reset

#### What does this PR do?
- Password reset.
- Refactor tests to use basetest file.
- Update email sending mechanism to use sendgrid API.

#### Description of Task to be completed?
Get the following endpoints working.
`POST api/user/reset_password`
`PUT user/reset_password/<reset_password_token>`

#### How should this be manually tested?
- Given the user forgets or wants to reset his/her password.
- In post man using the url:POST-api/user/reset_password, he/she enters the email address as the payload below.
````
{ "email" : "email-address" }
````
- Then an email is set to the given email address with the url: PUT-user/reset_password/<reset_password_token>
- Again in postman using the url in the email PUT the new password using the payload;
````
{
    "new_password": "jude1234*",
    "confirm_password": "jude1234*"
}
````
- After try out the new password on login to confirm that it has been reset.

#### What are the relevant pivotal tracker stories?
[#161348794](https://www.pivotaltracker.com/story/show/161348794)
